### ISSUE TITLE
How to use Alternatives?

### ISSUE THREAD

**Issue by codeslayer1** [whefter/nestjs-joi#2]
I want to use `alternatives` to validate my class variable against multiple schema. So for ex if I have a class A which has an array variable called items, I want to validate whether its elements match ones of these classes - Item1 or Item2.

Example

```
export class A {
   // doesn't work since Item1 and Item2 are classes and not Joi Schema
   // getting error - failed custom validation because Class constructor EmployeeAssetDto cannot be invoked without 'new'
   @JoiSchema(Joi.array().items(Joi.alternatives(Item1, Item2)))
   // this also doesn't work, works well with single schema
   @JoiSchema([Item1, Item2])
   items: Item1[] | Item2[]
}

export class Item1 {
   @JoiSchema(Joi.string().optional())
   id?: string;

   @JoiSchema(Joi.string().optional())
   status?: string;
}

export class Item2 {
   @JoiSchema(Joi.string().optional())
   id2?: string;

   @JoiSchema(Joi.string().optional())
   status2?: string;
}
```

Can you please point me in right direction to do this?

**Comment by whefter** [comment]
The module doesn't look into already constructed schemas, so it can't see that something is meant to be an alternative. A more advanced use case like this is not currently supported.

I'm assuming writing our the alternatives explicitely (`Joi.alternatives(Joi.object().keys({id1: ..}), Joi.object().keys({id2: ...})`) isn't an option here because you actually need the `Item1` and `Item2` DTOs in other places and you want to DRY?

Possibly providing the inner type-to-schema logic as an explicit function (`typeToSchema`) would be an option here:
```
Joi.alternatives(schemaFromType(Item1), schemaFromType(Item2));
```

I'll take a swing at that when I get a moment.

**Comment by whefter** [comment]
Done, available in v1.1.0. Should be published in a few minutes: https://github.com/whefter/nestjs-joi/blob/master/README.md#gettypeschematype

**Comment by codeslayer1** [comment]
Thanks. Exactly what I needed. Working fine in v1.1.
